WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java

```diff
@@ -49,4 +58,102 @@ public static int getWordCount(String content) {
         String text = Html.fromHtml(content.replaceAll("<img[^>]*>", "")).toString();
         return text.split("\\s+").length;
     }
+
+    /**
+     * Bump Analytics for the passed Stat and CURRENT blog details into properties.
+     *
+     * @param stat The Stat to bump
+     *
+     */
+    public static void trackWithCurrentBlogDetails(AnalyticsTracker.Stat stat) {
+        trackWithCurrentBlogDetails(stat, null);
+    }
+
+    /**
+     * Bump Analytics for the passed Stat and CURRENT blog details into properties.
+     *
+     * @param stat The Stat to bump
+     * @param properties Properties to attach to the event
+     *
+     */
+    public static void trackWithCurrentBlogDetails(AnalyticsTracker.Stat stat, Map<String, Object> properties) {
+        trackWithBlogDetails(stat, WordPress.getCurrentBlog(), properties);
+    }
+
+    /**
+     * Bump Analytics for the passed Stat and add blog details into properties.
+     *
+     * @param stat The Stat to bump
+     * @param blog The blog object
+     *
+     */
+    public static void trackWithBlogDetails(AnalyticsTracker.Stat stat, Blog blog) {
+        trackWithBlogDetails(stat, blog, null);
+    }
+
+    /**
+     * Bump Analytics for the passed Stat and add blog details into properties.
+     *
+     * @param stat The Stat to bump
+     * @param blog The blog object
+     * @param properties Properties to attach to the event
+     *
+     */
+    public static void trackWithBlogDetails(AnalyticsTracker.Stat stat, Blog blog, Map<String, Object> properties) {
+        if (blog == null || (!blog.isDotcomFlag() && !blog.isJetpackPowered())) {
+            AppLog.w(AppLog.T.STATS, "The passed blog obj is null or it's not a wpcom or Jetpack. Tracking analytics without blog info");
+            AnalyticsTracker.track(stat, properties);
+            return;
+        }
+
+        String blogID = StatsUtils.getBlogId(blog);
```

Comment:
We should probably move `StatsUtils.getBlogId` to another Util now it's used in other place than Stats. I'm also wondering if we should reuse it in other places where we need a remote blog id for .com/.org/jetpack site.